src/Common/Factory/AbstractFactoryRule.php

```diff
@@ -13,55 +13,33 @@
 
 abstract class AbstractFactoryRule extends SprykerAbstractRule
 {
-    public const PATTERN_FACTORY = '/\\\\*\\\\.+\\\\.+\\\\[A-Za-z0-9]+(Business|Service|Communication|Persistence)Factory$/';
-    public const PATTERN_FACTORY_EXCEPT_PERSISTENCE = '/\\\\*\\\\.+\\\\.+\\\\[A-Za-z0-9]+(Business|Service|Communication)Factory$/';
+    public const PATTERN_FACTORY = '/\\\\*\\\\.+\\\\.+\\\\[A-Za-z0-9]+(Business|Service|Communication)Factory$/';
 
     /**
      * @param \PHPMD\Node\AbstractNode $node
      *
      * @return bool
      */
-    protected function isFactory(AbstractNode $node): bool
+    protected function isFactory(AbstractNode $node)
     {
-        $className = $this->getClassName($node);
+        if ($node instanceof MethodNode) {
+            $node = $this->getNodeFromMethodNode($node);
 
-        if (preg_match(static::PATTERN_FACTORY, $className) === 0) {
-            return false;
+            $parent = $node->getParent();
+            $className = $parent->getNamespaceName() . '\\' . $parent->getName();
+        } else {
+            $className = $node->getFullQualifiedName();
         }
 
-        return true;
-    }
-
-    /**
-     * @param \PHPMD\Node\AbstractNode $node
-     *
-     * @return bool
-     */
-    protected function isFactoryExceptPersistence(AbstractNode $node): bool
-    {
-        $className = $this->getClassName($node);
-
-        if (preg_match(static::PATTERN_FACTORY_EXCEPT_PERSISTENCE, $className) === 0) {
+        // Factories in persistence layer have other rules
+        if (strpos($className, '\\Persistence\\') !== false) {
```

Comment:
why to they have other rules? I thought we agreed they dont?

Reply:
> Why not also just saying they only need to allow public methods?
That's what we talking about? Every rule except public methods checking has been reverted to the state it was before.